File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int prod = 1;
        int hasZero = 0;

        for(int i =0;i<nums.length;i++){
            if(nums[i] == 0) {
                hasZero++;
                continue;
                }
            
            prod = prod*nums[i];
        }

        if(hasZero > 1) prod =0;

        for(int i = 0;i<nums.length;i++){
            if(nums[i] == 0) {
                nums[i] = prod;
                continue;
            }
            if(hasZero > 0) nums[i] = 0;
            else
            nums[i] = prod/nums[i];
        }

        return nums;
    }
}  

File: Data Structures & Algorithms/binary-search/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let i = 0, l = nums.length-1;
        while(i<=l){
            if(nums[i]=== target) return i;
            if(nums[l]===target) return l;
            i++;
            l--;
        }

        return -1;
    }
}
